core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -321,7 +321,7 @@ private[spark] class Executor(
           logInfo(s"Executor killed $taskName (TID $taskId)")
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled))
 
-        case cDE: CommitDeniedException =>
+        case CausedBy(cDE: CommitDeniedException) =>
```

Comment:
@srowen this is useful in other places too, so I don't think it's overkill. What you suggested handles only 1 level of nesting and is less robust. I would prefer to leave this the way it is.